app/javascript/packages/form-steps/use-history-param.ts

```diff
@@ -2,30 +2,16 @@ import { useState, useEffect, useCallback } from 'react';
 
 export type ParamValue = string | undefined;
 
-interface HistoryOptions {
-  basePath?: string;
-}
-
 /**
  * Returns the step name from a given path, ignoring any subpaths or leading or trailing slashes.
  *
  * @param path Path from which to extract step.
  *
  * @return Step name.
  */
-export const getStepParam = (path: string): string =>
-  decodeURIComponent(path.split('/').filter(Boolean)[0]);
-
-export function getParamURL(value: ParamValue, { basePath }: HistoryOptions): string {
-  let prefix = typeof basePath === 'string' ? basePath.replace(/\/$/, '') : '#';
-  if (value && basePath) {
-    prefix += '/';
-  }
-
-  return [prefix, encodeURIComponent(value || '')].filter(Boolean).join('');
-}
+export const getStepParam = (path: string): string => decodeURIComponent(path.replace(/^#/, ''));
 
-const onURLChange = () => window.dispatchEvent(new window.CustomEvent('lg:url-change'));
+const getParamURL = (value: ParamValue) => `#${encodeURIComponent(value || '')}`;
```

Comment:
WDYT of naming this `buildParamURL` since `get` makes me think it's like extracting something instead of formatting it?

Reply:
> WDYT of naming this `buildParamURL` since `get` makes me think it's like extracting something instead of formatting it?

I have a soft preference to leave it as it is, or at least (despite what the diff would imply) this function already existed and wasn't something I'd aimed to change, aside from the implementation detail of considering the `basePath` parameter.